cdap-master/src/main/java/co/cask/cdap/data/tools/HBaseTableExporter.java

```diff
@@ -0,0 +1,212 @@
+      new ExploreClientModule(),
+      new ViewAdminModules().getDistributedModules(),
+      new StreamAdminModules().getDistributedModules(),
+      new NotificationFeedServiceRuntimeModule().getDistributedModules()
+    );
+  }
+
+  /**
+   * Sets up the actual MapReduce job.
+   * @param tx The transaction which needs to be passed to the Scan instance. This transaction is be used by
+   *           coprocessors to filter out the data corresonding to the invalid transactions .
+   * @param tableName Name of the table which need to be exported as HFiles.
+   * @return the configured job
+   * @throws IOException
+   */
+  public Job createSubmittableJob(Transaction tx, String tableName) throws IOException {
+
+    Job job = Job.getInstance(hConf, "HBaseTableExporter");
+
+    job.setJarByClass(HBaseTableExporter.class);
+    Scan scan = new Scan();
+    scan.setCacheBlocks(false);
+    // Set the transaction attribute for the scan.
+    scan.setAttribute(TxConstants.TX_OPERATION_ATTRIBUTE_KEY, new TransactionCodec().encode(tx));
+    job.setNumReduceTasks(0);
+
+    TableMapReduceUtil.initTableMapperJob(tableName, scan, Import.KeyValueImporter.class, null, null, job);
+
+    FileSystem fs = FileSystem.get(hConf);
+    Random rand = new Random();
+    Path root = new Path(fs.getWorkingDirectory(), "hbasetableexporter");
+    fs.mkdirs(root);
+    while (true) {
+      bulkloadDir = new Path(root, "" + rand.nextLong());
+      if (!fs.exists(bulkloadDir)) {
+        break;
+      }
+    }
+
+    HFileOutputFormat2.setOutputPath(job, bulkloadDir);
+    HTable hTable = new HTable(hConf, tableName);
+    HFileOutputFormat2.configureIncrementalLoad(job, hTable);
+
+    return job;
+  }
+
+  private void startUp() throws Exception {
+    zkClientService.startAndWait();
+    txService.startAndWait();
+  }
+
+  private void stop() throws Exception {
+    try {
+      txService.stopAndWait();
+      zkClientService.stopAndWait();
+    } catch (Throwable t) {
+      LOG.error("Exception while trying to stop the services.", t);
+      Runtime.getRuntime().halt(1);
+    }
+  }
+
+  private void printHelp() {
+    System.out.println();
+    System.out.println("Usage: /opt/cdap/master/bin/svc-master " +
+                         "run co.cask.cdap.data.tools.HBaseTableExporter <tablename>");
+    System.out.println();
+    System.out.println("Args:");
+    System.out.println(" tablename    Name of the table to copy");
+  }
+
+  public void doMain(String[] args) throws Exception {
+    if (args.length < 1) {
+      printHelp();
+      return;
+    }
+
+    String tableName = args[0];
+
+    try {
+      startUp();
+      Transaction tx = txClient.startLong();
+      Job job = createSubmittableJob(tx, tableName);
+      if (job == null) {
+        throw new RuntimeException("Failed to configure the job.");
+      }
+      if (!job.waitForCompletion(true)) {
+        LOG.info("MapReduce job failed!");
+        Transactions.invalidateQuietly(txClient, tx);
```

Comment:
Discussed this with @poornachandra and since we are not doing any data update operations in the MapReduce job, its ok to commit the transaction always. Updated the code accordingly.